crnk-core/src/main/java/io/crnk/core/engine/security/SecurityProvider.java

```diff
@@ -1,8 +1,10 @@
 package io.crnk.core.engine.security;
 
+import io.crnk.core.engine.result.Result;
+
 public interface SecurityProvider {
 
-	boolean isUserInRole(String role);
+	Result<Boolean> isUserInRole(String role);
```

Comment:
based on https://github.com/crnk-project/crnk-framework/issues/539#issuecomment-507581394 maybe we should rather have:

```
public interface SecurityProvider {
   boolean isUserInRole(SecurityProviderContext context, String role);
}

interface SecurityProviderContext {
   QueryContext getQueryContext();
}
```

issue is that all security related code is called quite extensively. So it must return immediately.

Reply:
alright. Reverted previous change and added the SecurityProviderContext.